sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/AadAuthenticationFilterAutoConfiguration.java

```diff
@@ -98,19 +101,29 @@ public AadAppRoleStatelessAuthenticationFilter aadStatelessAuthFilter(ResourceRe
         );
     }
 
+    /**
+     * Declare RestOperations bean used by various Nimbus JWT components.
+     *
+     * @param builderObjectProvider the optional rest template builder bean.
+     * @return RestOperations bean
+     */
+    @Bean
+    @ConditionalOnMissingBean(RestOperations.class)
+    public RestOperations aadAuthRestOperations(ObjectProvider<RestTemplateBuilder> builderObjectProvider) {
```

Comment:
I propose the bean `RestOperations` should be also taken by XXXAccessTokenResponseClients in the AAD OAuth2 client configuration because this will be not initiated when the application type is `web_application_and_resource_server`.

Reply:
These token response clients should take the bean `RestOperations` by default, they are created via class `DefaultJwtBearerTokenResponseClient`, `DefaultRefreshTokenTokenResponseClient`, `DefaultPasswordTokenResponseClient`, `DefaultClientCredentialsTokenResponseClient` in configuration `AadOAuth2ClientConfiguration`, [DefaultAuthorizationCodeTokenResponseClient](https://github.com/Azure/azure-sdk-for-java/blob/316164cf154ce24b317d92b02151783b66fcef00/sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/AadWebSecurityConfigurerAdapter.java#L123) in configuration `AadWebSecurityConfigurerAdapter`.

Reply:
Updated.